fred2/ignoreordersdlg.cpp

```diff
@@ -199,31 +154,32 @@ void ignore_orders_dlg::OnOK()
 	int i;
 	object *objp;
 
+	CCheckListBox* ignore_orders_list = ((CCheckListBox*)GetDlgItem(IDC_IGNORE_ORDERS_LIST));
+
 	// clear out the orders, then set the bits according to which check boxes are set
 	if ( m_ship >= 0 ) {
-		for ( i = 0; i < m_num_checks_active; i++ ) {
-			if ( check_boxes[i].button->GetCheck() )
-				orders_accepted.insert(check_boxes[i].id);
+		for ( i = 0; i < (int) m_orderList.size(); i++) {
+			if (ignore_orders_list->GetCheck(i) == BST_CHECKED)
+				orders_accepted.insert(m_orderList[i]);
 		}
 		Ships[m_ship].orders_accepted = orders_accepted;
 	} else {
 		for ( objp = GET_FIRST(&obj_used_list); objp != END_OF_LIST(&obj_used_list); objp = GET_NEXT(objp) ) {
 			if (((objp->type == OBJ_SHIP) || (objp->type == OBJ_START)) && (objp->flags[Object::Object_Flags::Marked])) {
-				Ships[objp->instance].orders_accepted.clear();
```

Comment:
The previous handling was simply incorrect here. It would clear all allowed orders, and then remove orders explicitly disallowed (from the empty set, thus doing nothing) and readding explicitly allowed entries. That leaves the indeterminate-state orders also cleared. This is wrong, however, as we want orders for which the checkboxes are indeterminate to stay as they are in their respective ships (which may or may not be allowed). To do that, we don't clear the orders and leave them as is, then make sure they're up to date by removing disallowed orders and readding allowed orders.